src/lib/__tests__/agents.test.ts

```diff
@@ -22,24 +35,121 @@ const bobConfig = {
+      .to(ContextImpl)
+      .inSingletonScope();
+    aliceContainer
+      .bind<ConnectionService>(TYPES.ConnectionService)
+      .to(ConnectionService)
+      .inSingletonScope();
+    aliceContainer.bind<BasicMessageService>(TYPES.BasicMessageService).to(BasicMessageService);
+    aliceContainer
+      .bind<ProviderRoutingService>(TYPES.ProviderRoutingService)
+      .to(ProviderRoutingService)
+      .inSingletonScope();
+    aliceContainer.bind<ConsumerRoutingService>(TYPES.ConsumerRoutingService).to(ConsumerRoutingService);
+    aliceContainer.bind<MessageReceiver>(TYPES.MessageReceiver).to(MessageReceiver);
+    ContainerHelper.registerDefaultHandlers(aliceContainer);
+    aliceContainer.bind<Dispatcher>(TYPES.Dispatcher).to(Dispatcher);
+
+    aliceContainer
+      .bind<Agent>(TYPES.Agent)
+      .to(Agent)
+      .inSingletonScope();
+    aliceAgent = aliceContainer.get<Agent>(TYPES.Agent);
+
+    bobContainer = new Container();
+    bobContainer
+      .bind<Subject<unknown>>(TYPE_SUBJECT)
+      .toConstantValue(bobMessages)
+      .whenTargetNamed('receiver');
+    bobContainer
+      .bind<Subject<unknown>>(TYPE_SUBJECT)
+      .toConstantValue(aliceMessages)
+      .whenTargetNamed('sender');
+    bobContainer
+      .bind<OutboundTransporter>(TYPES.OutboundTransporter)
+      .to(SubjectOutboundTransporter)
+      .inSingletonScope();
+    bobContainer
+      .bind<InboundTransporter>(TYPES.InboundTransporter)
+      .to(SubjectInboundTransporter)
+      .inSingletonScope();
+    bobContainer.bind<InitConfig>(TYPES.InitConfig).toConstantValue(bobConfig);
+    bobContainer.bind<WalletConfig>(TYPES.WalletConfig).toConstantValue({ id: bobConfig.walletName });
+    bobContainer.bind<WalletCredentials>(TYPES.WalletCredentials).toConstantValue({ key: bobConfig.walletKey });
+    bobContainer
+      .bind<Wallet>(TYPES.Wallet)
+      .to(IndyWallet)
+      .inSingletonScope();
+    bobContainer.bind<MessageSender>(TYPES.MessageSender).to(MessageSender);
+    bobContainer
+      .bind<Context>(TYPES.Context)
+      .to(ContextImpl)
+      .inSingletonScope();
+    bobContainer
+      .bind<ConnectionService>(TYPES.ConnectionService)
+      .to(ConnectionService)
+      .inSingletonScope();
+    bobContainer.bind<BasicMessageService>(TYPES.BasicMessageService).to(BasicMessageService);
+    bobContainer
+      .bind<ProviderRoutingService>(TYPES.ProviderRoutingService)
+      .to(ProviderRoutingService)
+      .inSingletonScope();
+    bobContainer.bind<ConsumerRoutingService>(TYPES.ConsumerRoutingService).to(ConsumerRoutingService);
+    bobContainer.bind<MessageReceiver>(TYPES.MessageReceiver).to(MessageReceiver);
+    ContainerHelper.registerDefaultHandlers(bobContainer);
+    bobContainer.bind<Dispatcher>(TYPES.Dispatcher).to(Dispatcher);
+
+    bobContainer
+      .bind<Agent>(TYPES.Agent)
+      .to(Agent)
+      .inSingletonScope();
+    bobAgent = bobContainer.get<Agent>(TYPES.Agent);
+  });
```

Comment:
I think our goal should be to provide a library ready to use without needing much to configure and the e2e test should demonstrate this. It means to hide library internals, expose just what's necessary and allow a developer to extend or modify functionality via plugins. For example, a developer would be able to define what transport protocol, storage and aries protocols she wants to use, but she shouldn't need to know anything about Dispatcher, MessageSender, MessageReceiver and so on.

Reply:
True. I reckon a decent set of defaults can always be added on to a helper method in `ContainerHelper.ts` and the user can just get them all in by calling that method while setting up the container as I did with the handlers